src/Symfony/Component/Serializer/Normalizer/BackedEnumNormalizer.php

```diff
@@ -57,6 +57,14 @@ public function denormalize($data, string $type, ?string $format = null, array $
             throw new InvalidArgumentException('The data must belong to a backed enumeration.');
         }
 
+        if ($data instanceof $type) {
+            return $data;
+        }
+
+        if ($data instanceof \BackedEnum) {
+            $data = $data->value;
```

Comment:
converting a backed enum from a different type first to its backing value and then hydrating the expected type from it makes even less sense to me than what the PR description is describing. This is definitely a no-go from me.